mimic-iii/buildmimic/duckdb/import_duckdb.py

```diff
@@ -0,0 +1,378 @@
+                sql = fp.read()
+                result = connection.execute(sql).fetchall()
+                for row in result:
+                    print(f"{row[0]}: {row[2]} records ({row[1]} expected) - {row[3]}")
+
+        except Exception as error:
+            print("Failed setting up database: ", error)
+            raise error
+        finally:
+            if connection:
+                connection.close()
+                print("duckdb connection is closed")
+
+    #TODO: If both --schema-name and --skip-tables are specified, we won't have
+    # populated tables_in_schema with the data table names... so the views won't
+    # work... So, here, read the tables already in the destination schema from
+    # the DB and add those tablenames to tables_in_schema?
+
+    if make_concepts:
+        connection = duckdb.connect(output_db)
+        print("Connected to duckdb...")
+
+        print("Creating tables...")
+
+        # ccs_dx is an outlier...this is adapted from the BigQuery version...
+        ccs_multi_dx_create = """
+            DROP TABLE IF EXISTS ccs_multi_dx;
+            CREATE TABLE ccs_multi_dx
+            (
+            icd9_code CHAR(5) NOT NULL,
+            -- CCS levels and names based on position in hierarchy
+            ccs_level1 VARCHAR(10),
+            ccs_group1 VARCHAR(100),
+            ccs_level2 VARCHAR(10),
+            ccs_group2 VARCHAR(100),
+            ccs_level3 VARCHAR(10),
+            ccs_group3 VARCHAR(100),
+            ccs_level4 VARCHAR(10),
+            ccs_group4 VARCHAR(100)
+            );
+            """
+
+        print("Loading data...")
+        try:
+
+            connection.execute(f"USE main;")
+
+            connection.execute(ccs_multi_dx_create)
+            csvgz_path = os.path.join(mimic_code_root, 'mimic-iii','concepts_postgres','diagnosis','ccs_multi_dx.csv.gz')
+            connection.execute(f"COPY ccs_multi_dx from '{csvgz_path}' (FORMAT CSV, DELIMITER ',', HEADER);")
+            
+        except Exception as error:
+            print("Failed to setup ccs_multi_dx: ", error)
+            raise error
+        finally:
+            if connection:
+                connection.close()
+                print("duckdb connection is closed")
+
+        connection = duckdb.connect(output_db)
+
+        print("Creating views...")
+        try:
+
+            if schema_name is not None:
+                connection.execute(f"CREATE SCHEMA IF NOT EXISTS {schema_name};")
+                connection.execute(f"USE {schema_name}")
+
+            for key in concept_name_map:
+                if schema_name is not None:
+                    if "schema" not in concept_name_map[key]:
+                        tables_in_schema.add(key.lower())
+
+                #cProfile.run('...')
+                #print(f"Making view {key}...")
+                db = concept_name_map[key].get("db", "bigquery")
+                if db == "duckdb":
```

Comment:
I think this is unnecessarily abstract. You might as well just hard-code the make_duckdb_query function to do something special for icustay_hours. Or, alternatively, have the subfunction contain a hard-coded list of queries which it parses as duckdb, that way you can expand it to more queries in the future if you plan to move more over

Reply:
Yes.... earlier I was pulling some queries from the BigQuery version and others from the Postgres version, and Postgres was actually the default...later I got them all to work translated from BigQuery except this one...which now looks odd.